Reported on distributions@kde,

Due to a change in kdeconnect ioslave (changed it's KProtocol Class
to ":local"), it stopped working with Dolphin, i.e. accessing a
device from kdeconnect:// wouldn't work.  This issue has been fixed
in KIO, but the patches didn't make it to the 5.74 release.

diff --git a/src/widgets/kurlcompletion.cpp b/src/widgets/kurlcompletion.cpp
index 88bbaace3e2819af5541f5e2bedf073892ab374a..849a0f3872f03b6fa759f9d3ab90aae67d568cbb 100644
--- src/widgets/kurlcompletion.cpp
+++ src/widgets/kurlcompletion.cpp
@@ -1213,9 +1213,14 @@ void KUrlCompletionPrivate::_k_slotEntries(KIO::Job *, const KIO::UDSEntryList &
             entry_name = entry.stringValue(KIO::UDSEntry::UDS_NAME);
         }
 
-        //qDebug() << "name:" << name;
+        // This can happen with kdeconnect://deviceId as a completion for kdeconnect:/,
+        // there's no fileName [and the UDS_NAME is unrelated, can't use that].
+        // This code doesn't support completing hostnames anyway (see addPathToUrl below).
+        if (entry_name.isEmpty()) {
+            continue;
+        }
 
-        if ((!entry_name.isEmpty() && entry_name.at(0) == QLatin1Char('.')) &&
+        if (entry_name.at(0) == QLatin1Char('.') &&
                 (list_urls_no_hidden ||
                  entry_name.length() == 1 ||
                  (entry_name.length() == 2 && entry_name.at(1) == QLatin1Char('.')))) {
